packages/forms/src/model.ts

```diff
@@ -671,6 +671,71 @@ export abstract class AbstractControl {
     this._updateControlsErrors(opts.emitEvent !== false);
   }
 
+  /**
+   * Adds an error on a form control when running validations manually,
+   * rather than automatically.
+   *
+   * Calling `addError` also updates the validity of the parent control.
+   *
+   * @usageNotes
+   * ### Manually add an error for a control
+   *
+   * ```
+   * const login = new FormControl('someLogin');
+   * login.addError('notUnique', true);
+   * login.addError('tooComplicated', 'errorValue');
+   *
+   * expect(login.valid).toEqual(false);
+   * expect(login.errors).toEqual({ notUnique: true, tooComplicated: 'errorValue' });
+   *
+   * login.setValue('someOtherLogin');
+   *
+   * expect(login.valid).toEqual(true);
+   * ```
+   */
+  addError(error: string, errorValue: any, opts: {emitEvent?: boolean} = {}): void {
+    const validationErrors = (this as{errors: ValidationErrors | null}).errors;
+    if (validationErrors) {
+      validationErrors[error] = errorValue;
+      this.setErrors(validationErrors, opts);
+    } else {
+      this.setErrors({[error]: errorValue}, opts);
+    }
+  }
+
+  /**
+   * Removes an error on a form control when running validations manually,
+   * rather than automatically.
+   *
+   * Calling `removeError` also updates the validity of the parent control.
+   *
+   * @usageNotes
+   * ### Manually remove an error for a control
+   *
+   * ```
+   * const login = new FormControl('someLogin');
+   * login.setErrors({
+   *   notUnique: true,
+   *   tooComplicated: 'errorValue'
+   * });
+   *
+   * expect(login.valid).toEqual(false);
+   * expect(login.errors).toEqual({ notUnique: true, tooComplicated: 'errorValue' });
+   *
+   * login.removeError('tooComplicated');
+   *
+   * expect(login.valid).toEqual(false);
+   * expect(login.errors).toEqual({ notUnique: true });
+   * ```
+   */
+  removeError(error: string, opts: {emitEvent?: boolean} = {}): void {
+    const validationErrors = (this as{errors: ValidationErrors | null}).errors;
+    if (validationErrors) {
+      delete validationErrors[error];
+    }
+    this.setErrors(validationErrors, opts);
```

Comment:
In the case where validationErrors only contains the error being removed, this will set this.errors = {} as opposed to null, which will cause validation problems (since it will technically still be invalid).  I would suggest testing to see if the newly updated validationErrors is empty and if so setting the errors to null, maybe something like:

```suggestion
    const validationErrors = (this as{errors: ValidationErrors | null}).errors;
    let isEmpty;
    if (validationErrors) {
      delete validationErrors[error];
      isEmpty = Object.keys(validationErrors).length === 0;
    }
    this.setErrors(isEmpty ? null : validationErrors, opts);
```

Reply:
You're right, I updated 👍